mfai/pytorch/models/llms/fuyu.py

```diff
@@ -114,160 +117,81 @@ def __init__(
 
         # Builds linear projection layer for weather input data (same for each time step)
         # lat_dim, lon_dim, timestep_dim, features_dim
-
-        downsampled_lat = math.floor(
-            settings.vision_input_shape[0] / settings.downsampling_rate
-        )
-        downsampled_lon = math.floor(
-            settings.vision_input_shape[1] / settings.downsampling_rate
-        )
-        spatial_dims = downsampled_lat * downsampled_lon
-
-        # downsampled spatial dims
-        self.downsampler = nn.MaxPool2d(settings.downsampling_rate)
+        if settings.layer_norm_vis_txt:
+            self.norm_or_ident: nn.Identity | nn.LayerNorm = nn.LayerNorm(
+                self.settings.emb_dim
+            )
+        else:
+            self.norm_or_ident = nn.Identity()
 
```

Comment:
I suggest to define the type of `self.vision_encoder` before the if statement
```suggestion

self.vision_encoder: WeatherProjector | ResNet50MLM | VitEncoder
```

Reply:
ok !